spec/requests/api/blueprints_spec.rb

```diff
@@ -0,0 +1,196 @@
+
+      expect(response.parsed_body).to include("href" => a_string_matching(blueprints_url(blueprint.id)))
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "forbids access to a blueprint without an appropriate role" do
+      blueprint = FactoryGirl.create(:blueprint)
+      api_basic_authorize
+
+      run_get(blueprints_url(blueprint.id))
+
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  describe "POST /api/blueprints" do
+    it "can create a blueprint" do
+      api_basic_authorize collection_action_identifier(:blueprints, :create)
+
+      run_post(blueprints_url, :name => "foo", :description => "bar", :ui_properties => {:some => {:json => "baz"}})
+
+      expected = {
+        "results" => [
+          a_hash_including(
+            "name"          => "foo",
+            "description"   => "bar",
+            "ui_properties" => {"some" => {"json" => "baz"}}
+          )
+        ]
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "rejects blueprint creation with id specified" do
+      api_basic_authorize collection_action_identifier(:blueprints, :create)
+
+      run_post(blueprints_url, :name => "foo", :description => "bar", :id => 123)
+
+      expected = {
+        "error" => a_hash_including(
+          "kind"    => "bad_request",
+          "message" => a_string_matching(/id or href should not be specified/i),
+        )
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:bad_request)
+    end
+
+    it "rejects blueprint creation with an href specified" do
+      api_basic_authorize collection_action_identifier(:blueprints, :create)
+
+      run_post(blueprints_url, :name => "foo", :description => "bar", :href => blueprints_url(123))
+
+      expected = {
+        "error" => a_hash_including(
+          "kind"    => "bad_request",
+          "message" => a_string_matching(/id or href should not be specified/i),
+        )
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:bad_request)
+    end
+
+    it "can create a blueprint with a bundle by href with an appropriate role" do
+      service_dialog = build_dialog
+      service_template_1, service_template_2 = FactoryGirl.create_list(:service_template, 2)
+      service_catalog = FactoryGirl.create(:service_template_catalog)
+      api_basic_authorize collection_action_identifier(:blueprints, :create)
+
+      run_post(
+        blueprints_url,
+        :name        => "foo",
+        :description => "bar",
+        :bundle      => {
+          :service_catalog      => {:href => service_catalogs_url(service_catalog.id)},
+          :service_dialog       => {:href => service_dialogs_url(service_dialog.id)},
+          :automate_entrypoints => {"Provision" => "a/b/c", "Reconfigure" => "x/y/z"},
+          :service_templates    => [
```

Comment:
Are service_templates the items/nodes that are dragged onto a Blueprint's canvas in SSUI?